app/models/publisher.rb

```diff
@@ -55,6 +55,20 @@ class Publisher < ApplicationRecord
 
   scope :created_recently, -> { where("created_at > :start_date", start_date: 1.week.ago) }
 
+  # publishers that have uphold codes that have been sitting for five minutes
+  # can be cleared if publishers do not create wallet within 5 minute window
+  scope :has_stalled_uphold_code, -> {
+    where.not(encrypted_uphold_code: nil)
+    .where("uphold_updated_at < ?", Time.now - 5.minutes)
+  }
+
+  # publishers that have access params that havent accepted by eyeshade
+  # can be cleared after 2 hours
+  scope :has_stalled_uphold_access_parameters, -> {
+    where.not(encrypted_uphold_access_parameters: nil)
+    .where("uphold_updated_at < ?", Time.now - 2.hours)
+  }
```

Comment:
These scopes are tested indirectly with the job. Perhaps unit test them as well?

Reply:
That's a good idea, I will add them.  I'll also add some tests that make sure a publisher can  successfully acquire new uphold codes and access params if they have already been cleared.